### PR TITLE
Add endpoint to trigger plugin jobs

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

Pair PR to https://github.com/PostHog/plugin-server/pull/537

Reasonably generic endpoint to allow us to expand its capabilities in the future. 

Purely backend change, nothing yet in the frontend to leverage this endpoint (coming soon).

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Review by Twixes — COMMENTED** [review]
Good overall

File: posthog/api/plugin.py [hunk 1]
```diff
@@ -321,6 +326,29 @@ def rearrange(self, request: request.Request, **kwargs):
...
+
+        if "name" not in job:
+            raise ValidationError("A job name must be specified!")
+
+        job_name = job.get("name")
+        job_payload = job.get("payload", {})
+        job_op = job.get("operation", "start")
+
+        task_name = CELERY_PLUGIN_JOBS_TASK
+        celery_queue = settings.PLUGINS_CELERY_QUEUE
```
[the remark this answers is not in the document]
Comment (yakkomajuri) [reply posthog/api/plugin.py:345]: yeah copied over from `capture.py` tbf

**Comment by yakkomajuri** [comment]
Oh I was wondering where my branch with the frontend changes went hahah

Guess I committed stuff here, will get it out

**Comment by yakkomajuri** [comment]
Addressed all comments @Twixes

**Review by Twixes — CHANGES_REQUESTED** [review]

File: posthog/settings.py [hunk 2]
```diff
@@ -637,3 +637,5 @@ def show_toolbar(request):
         "statsd": {"handlers": ["console"], "level": "WARNING", "propagate": True,},
     },
 }
+
+CELERY_PLUGIN_JOBS_TASK = "posthog.tasks.plugins.plugin_job"
```
[the remark this answers is not in the document]
Comment (yakkomajuri) [reply posthog/settings.py]: honestly, there's no reason why this shouldn't be inlined. got rid of this as a "setting". Not like it should change or be referenced anywhere anyway, and if that happens we can change it.

**Comment by yakkomajuri** [comment]
✔️  @Twixes

**Review by Twixes — APPROVED** [review]
That was Sonic fast :shipit:

**Pull request merged**